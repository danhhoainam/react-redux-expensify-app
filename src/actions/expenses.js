import uuid from 'uuid';

// ADD EXPENSE
export const addExpense = (
    {
        description = '',
        note = '',
        amount = 0,
        createAt = 0
    } = {}
) => ({
    type: 'ADD_EXPENSE',
    expense: {
        id: uuid(),
        description,
        note,
        amount,
        createAt
    }
});
// EDIT EXPENSE
export const editExpense = (id, updates) => ({
    type: 'EDIT_EXPENSE',
    id,
    updates
});
// REMOVE EXPENSE
export const removeExpense = ({ id } = {}) => ({
    type: 'REMOVE_EXPENSE',
    id
});