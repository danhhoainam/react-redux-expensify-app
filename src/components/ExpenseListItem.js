import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import moment from 'moment';
import numeral from 'numeral';

const ExpenseListItem = ({id, description, amount, createAt, note}) => (
    <div>
        <h3><Link to={`/edit/${id}`}>{description}</Link></h3>
        <p>
            {numeral(amount / 100).format('$0,0.00')} 
            -  
            {moment(createAt).format('MMMM Do, YYYY')}
        </p>
        
    </div>
);

export default ExpenseListItem;