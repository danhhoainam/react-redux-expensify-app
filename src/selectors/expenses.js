import moment from 'moment';

// filters
const getVisibleExpenses = (expenses, { text, sortBy, startDate, endDate }) => {
    return expenses.filter((expense) => {
        const createAtMoment = moment(expense.createAt);
        const startDateMatch = startDate ? startDate.isSameOrBefore(createAtMoment, 'day') : true;
        const endDateMatch = endDate ? endDate.isSameOrAfter(createAtMoment, 'day') : true;
        const textMatch = expense.description.toLowerCase().includes(text.toLowerCase());

        return startDateMatch && endDateMatch && textMatch;
    }).sort((a, b) => {
        if (sortBy === 'date') {
            return a.createAt - b.createAt <= 0;
        } else if (sortBy === 'amount') {
            return a.amount - b.amount <= 0;
        }
    });
};

export default getVisibleExpenses;