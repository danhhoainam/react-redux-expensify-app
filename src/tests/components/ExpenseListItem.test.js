import React from 'react';
import ExpenseListItem from './../../components/ExpenseListItem';
import { shallow } from 'enzyme';
import expenses from './../fixture/expenses';

test('should render ExpenseListItem with default data', () => {
    const wrapper = shallow(<ExpenseListItem {...expenses[0]}/>);
    expect(wrapper).toMatchSnapshot();
});