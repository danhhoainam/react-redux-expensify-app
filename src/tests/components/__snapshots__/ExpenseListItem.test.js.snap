// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseListItem with default data 1`] = `
<div>
  <h3>
    <Link
      replace={false}
      to="/edit/1"
    >
      test
    </Link>
  </h3>
  <p>
    $1.50
    -
    January 1st, 1970
  </p>
</div>
`;
